### ISSUE TITLE
Usage guide (missing docs)

### ISSUE THREAD

**Issue by glensc** [jdalrymple/gitbeaker#3285]
**Description**

Really stupid question:

- where is the documentation and usage docs?

I tried typical places:
- readme in project repository: https://github.com/jdalrymple/gitbeaker -> no usage really
- npm package readme of @gitbreaker/cli -> contains only instruction how to import
- project link in github: https://github.com/jdalrymple/gitbeaker -> none present
- docs folder in repository: https://github.com/jdalrymple/gitbeaker it contains only contributing and faq
- faq in repository: https://github.com/jdalrymple/gitbeaker (btw the `Requester.ts` there link is 404)
- also looked open issues
- searched issues with "missing docs"

anyway, I deducted from typescript typing I should create the object like this:

```ts
    const api = new Gitlab({
      host,
      token,
    });
```

but that yields error:
```
node_modules/@gitbeaker/requester-utils/dist/index.js:98
      throw new ReferenceError("requesterFn must be passed");
            ^
ReferenceError: requesterFn must be passed
    at new BaseResource (node_modules/@gitbeaker/requester-utils/dist/index.js:98:13)
    at new Gitlab (node_modules/@gitbeaker/core/dist/index.js:6767:5)
```

but `requesterFn` is optional for Gitlab constructor... so what's going on?

could not find any `requesterFn`  documentation except faq mentions the parameter, but no details what it should be.

- @gitbeaker/core: 38.9.0
- @gitbeaker/requester-utils: 38.9.0

**Checklist**
- [ ] I have checked that this is not a duplicate issue.
- [ ] I have read the documentation.

**Comment by jdalrymple** [comment]
https://github.com/jdalrymple/gitbeaker/blob/main/packages/rest/README.md#usage

Hehe you were close

**Comment by jdalrymple** [comment]
I'll improve the clarity for the redirection. Also if you look at the npm package itself on npm it should be pointing to it's individual readme

**Comment by glensc** [comment]
so, I needed `@gitbraker/rest` package rather `@gitbraker/core`?

is there some (brief) guide for `gitlab` -> `@gitbraker` change?

but hey, don't closer too soon, actual bugs are still there:
1. `Gitlab` from `@gitbraker/core` is required but typing says it's optional
2. `Requester.ts` link in faq is 404

**Comment by jdalrymple** [comment]
Correct the core library is if you wanted to build something off the API itself using custom request logic. It's not really for consumption for 99% of usecases. Use the rest library.

Not sure I understand your 1. point, did you mean requester ?

2. I will fix.

I closed the issue because the two points you mentioned have nothing to do with not finding the readme 😜. I will create a PR for the typing typo and broken link though!

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`38.10.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/38.10.0) :rocket:
